lib/chai/core/assertions.js

```diff
@@ -537,10 +537,11 @@ module.exports = function (chai, _) {
         itemsCount = val.length;
         break;
       case 'function':
-        throw new TypeError('.empty() was passed a function');
+        var name = val.name ? ' ' + val.name : '';
+        throw new TypeError('.empty was passed a function' + name);
       default:
         if (val !== Object(val)) {
-          throw new TypeError('.empty() was passed non-string primitive');
+          throw new TypeError('.empty was passed non-string primitive ' + String(val));
```

Comment:
Okay, so we're going for `inspect`.
Also, since `inspect` also adds delimiters which denote the type of the value (like `[]` for arrays and `''` for strings), it is more complete.
Thanks for this @shvaikalesh!

LGTM!